packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -0,0 +1,829 @@
+
+    /// @notice Mapping of operation IDs to the adapter that processed them
+    mapping(bytes32 operationId => address adapterAddress)
+        public operationToAdapter;
+
+    /// @notice Mapping of request IDs to the adapter that processed them
+    mapping(bytes32 requestId => address receivingAdapter)
+        public requestReceivedByAdapter;
+
+    /// @notice Mapping to track read request originators
+    mapping(bytes32 requestId => address originator)
+        public readRequestToOriginator;
+
+    /// @notice Pause state of the router
+    bool public paused;
+
+    /// @notice Mapping of chain IDs to their BridgeRouter addresses
+    mapping(uint16 chainId => address routerAddress)
+        public chainToRouterAddress;
+
+    /// @notice Address of the associated BridgeQueue
+    address public bridgeQueue;
+
+    /*//////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Initializes the BridgeRouter contract
+     * @param accessManager Address of the ProtocolAccessManager contract
+     * @param _bridgeQueue Address of the BridgeQueue contract
+     */
+    constructor(
+        address accessManager,
+        address _bridgeQueue
+    ) ProtocolAccessManaged(accessManager) {
+        bridgeQueue = _bridgeQueue;
+        emit BridgeQueueUpdated(_bridgeQueue);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        MODIFIERS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Modifier ensuring the caller (`msg.sender`) is a registered adapter.
+     * Reverts with `UnknownAdapter` if the caller is not in the `adapters` set.
+     */
+    modifier onlyRegisteredAdapter() {
+        if (!adapters.contains(msg.sender)) revert UnknownAdapter();
+        _;
+    }
+
+    /**
+     * @dev Modifier ensuring the caller (`msg.sender`) is the configured `bridgeQueue`.
+     * Reverts with `OnlyBridgeQueue` if the caller is not the `bridgeQueue` address.
+     */
+    modifier onlyBridgeQueue() {
+        if (msg.sender != bridgeQueue) revert OnlyBridgeQueue();
+        _;
+    }
+
+    /**
+     * @dev Modifier ensuring the contract is not paused.
+     * Reverts with `Paused` if the contract is in the paused state.
+     */
+    modifier whenNotPaused() {
+        if (paused) revert Paused();
+        _;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                       INTERNAL UTILITY FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Internal function to validate transfer parameters
+     * @param params Parameters to validate
+     */
+    function _validateTransferParams(
+        BridgeTypes.ExecuteTransferParams calldata params
+    ) internal pure {
+        if (
+            params.amount == 0 ||
```

Comment:
should that be done on bridge router level or at the adapter level - to validate available liquidty ec